SignalMessaging/contacts/OWSContactsManager.m

```diff
@@ -36,6 +36,7 @@ @interface OWSContactsManager () <SystemContactsFetcherDelegate>
 @property (nonatomic, readonly) SystemContactsFetcher *systemContactsFetcher;
 @property (nonatomic, readonly) YapDatabaseConnection *dbReadConnection;
 @property (nonatomic, readonly) YapDatabaseConnection *dbWriteConnection;
+@property (nonatomic, readonly) AnyLRUCache *cnContactCache;
```

Comment:
We use `AnyLRUCache` instead of `NSCache` because it is more aggressive about enforcing cache size, which turns out to matter.

Reply:
I haven't seen any evidence that NSCache is problematic here. Previously we just were **never** evicting the CNContacts because they were strongly referenced by our list of `OWSContactManager#allContacts: [Contact]`.

So unless you know something I don't, I'd say we should stick with NSCache here unless we have a strong reason not to. NSCache gives us more flexibility to have a larger cache size when possible, while also having a smaller cache size when the system is under pressure.

Reply:
I've done a bunch of measuring.  I thought I had a "smoking gun" scenario where `LRUCache` dramatically out-performs `NSCache`, but I can't remember/find it now.  I'll go with `NSCache`.